lib/Endpoint.js

```diff
@@ -39,6 +40,31 @@ class Endpoint extends EventEmitter {
     });
   }
 
+  /**
+   * Checks if `clusterName` is an `inputCluster` on this endpoint.
+   * @param {string} clusterName - Cluster#NAME
+   * @returns {boolean}
+   */
+  hasInputCluster(clusterName) {
+    if (typeof clusterName !== 'string') throw new TypeError('Error: `clusterName` must be string');
+    const cluster = Cluster.getCluster(clusterName) || {};
+    const clusterId = cluster.ID;
+    return (this._descriptor.inputClusters || []).includes(clusterId);
+  }
```

Comment:
I like the symmetry of `hasInputCluster` and `hasOutputCluster` but it makes `this.clusters` a bit strange.

This suggestion depends entirely on how many other components depend on this but you could make rename `this.clusters` to `this.inputClusters` and add `this.outputClusters`. Then the implementation of `hasInputCluster` (for example) could just be:

```js
hasInputCluster(clusterName) {
  if (typeof clusterName !== 'string') throw new TypeError('Error: `clusterName` must be string');
  return Boolean(this.inputClusters[clusterName]);
}
```

alternatively `Object.keys(this.inputClusters).includes(clusterName);` or if you can make it a `Map` `this.inputClusters.has(clusterName)`.

Just silly suggestions, I fully expect that breaking the api (renaming `this.clusters`) is not possible/desirable at this point in which case the code is great as is!